tests/api/retrieval.test.js

```diff
@@ -0,0 +1,262 @@
+      if (sql.includes("SELECT bearer_token as bearerToken FROM templates")) {
+        return {
+          rows: [{ bearerToken: token }],
+          rowCount: 1,
+        };
+      }
+    });
+    const dynamodbExpectedReponses = {
+      Items: [
+        { FormID: "1", SubmissionID: "12", FormSubmission: true },
+        { FormID: "2", SubmissionID: "21", FormSubmission: true },
+      ],
+    };
+    dynamoClient.send.mockImplementation(() => {
+      return dynamodbExpectedReponses;
+    });
+    DynamoDBClient.mockReturnValue(dynamoClient);
+    await retrieval(req, res);
+    expect(res.statusCode).toBe(200);
+    expect(JSON.parse(res._getData())).toEqual(
+      expect.objectContaining({
+        responses: [
+          { FormID: "1", SubmissionID: "12", FormSubmission: true },
+          { FormID: "2", SubmissionID: "21", FormSubmission: true },
+        ],
+      })
+    );
+  });
+
+  it("Should return 500 status code whenever dynamodb throws an error", async () => {
+    const mockSession = {
+      expires: "1",
+      user: { email: "forms@cds.ca", name: "forms" },
+    };
+    const token = jwt.sign(
+      {
+        formID: 2,
+      },
+      process.env.TOKEN_SECRET,
+      {
+        expiresIn: "1y",
+      }
+    );
+    client.getSession.mockReturnValue(mockSession);
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000/api/retrieval?maxRecords=5",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "5",
+      },
+    });
+    executeQuery.mockImplementation((client, sql) => {
+      if (sql.includes("SELECT bearer_token as bearerToken FROM templates")) {
+        return {
+          rows: [{ bearerToken: token }],
+          rowCount: 1,
+        };
+      }
+    });
+    DynamoDBClient.mockReturnValue(dynamoClient);
+    dynamoClient.send.mockImplementation(() => {
+      throw new Error("Error");
+    });
+    await retrieval(req, res);
+    expect(res.statusCode).toBe(500);
+  });
+
+  it("Should return an empty list of form's responses with 200 status code", async () => {
+    const mockSession = {
+      expires: "1",
+      user: { email: "forms@cds.ca", name: "forms" },
+    };
+    const token = jwt.sign(
+      {
+        formID: 3,
+      },
+      process.env.TOKEN_SECRET,
+      {
+        expiresIn: "1y",
+      }
+    );
+    client.getSession.mockReturnValue(mockSession);
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000/api/retrieval?maxRecords=10",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "10",
+      },
+    });
+    executeQuery.mockImplementation((client, sql) => {
+      if (sql.includes("SELECT bearer_token as bearerToken FROM templates")) {
```

Comment:
This will need to be changed to `'SELECT bearer_token as "bearerToken" FROM templates'` when merged with the PR that I currently have up.

Reply:
I've written a new validator where i no longer use this query.